[skia,crashpad,zeroc-ice,vcpkg-tool-gn] New vcpkg-tool-gn and update of several ports

bold84 opened the pull request:
**Describe the pull request**

- #### What does your PR fix?
  Updates Skia to newer version

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  unchanged

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

bold84 committed:
newer Skia version

JonLiu1993 commented on ports/skia/portfile.cmake:
```diff
@@ -1,8 +1,8 @@
-    REF fb0b35fed5580d49392df7ce9374551b348fffbf
-    PATCHES add-missing-tuple.patch
+    REF 1a668363fdf0d663c6df3f124e7c10796bc842f0
+    #PATCHES add-missing-tuple.patch
```
@bold84 ,Thanks for posting this issue, could you clarify why remove this patch?

bold84 replied:
Actually.. I just realized that the workspace wasn't clean when I updated the version. The opengl feature was also not intended to be part of this PR. 
I will look into this to find out why I disabled this PR.

bold84 replied:
The patches have been removed because they fail. And I have seen somewhere that these changes have already been merged to skia. But I can't find anymore where. :-(

JonLiu1993 commented on ports/skia/portfile.cmake:
```diff
@@ -108,6 +108,7 @@ skia_use_lua=false \
 skia_enable_tools=false \
 skia_enable_spirv_validation=false")
 
+
```
```suggestion
```

bold84 committed:
Update ports/skia/portfile.cmake

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

JonLiu1993 commented:
@bold84 , If this pr is ready for review please let me know, thanks!

bold84 committed:
use absolute python 3 path

bold84 committed:
fix issue with gn file

bold84 committed:
Update GN for Windows

bold84 committed:
updated gn for mac and linux (amd64 / arm64)

bold84 committed:
removed empty line

bold84 commented:
This depends on https://github.com/microsoft/vcpkg/pull/24142

bold84 committed:
Remove patch files

bold84 committed:
changes from vejmartin

bold84 committed:
make skia build x64-windows

bold84 committed:
update version

bold84 committed:
update version

bold84 committed:
enable linux

bold84 committed:
update crashpad

bold84 committed:
make crashpad build on windows

bold84 committed:
formatted crashpad/vcpkg.json

bold84 committed:
update version

bold84 commented:
This is weird... I don't have these errors on two Windows machines with Windows 11 and VS 2022. 

What's even weirder is that even the zeroc-ice port (x64-windows-static) suddenly fails on your windows servers, although nothing was changed. 

<img width="997" alt="Screen Shot 2022-04-16 at 03 02 56" src="https://user-images.githubusercontent.com/21118257/163626746-1d584486-b4c8-4ed7-857f-10b4b66c9ef5.png">

bold84 committed:
remove portfile

bold84 committed:
add file again

bold84 committed:
Looks like the CMake's rules for finding libraries have changed between cmake version 3.22.22011901-MSVC_2 and cmake version 3.22.22022201-MSVC_2 . :-D

At least I don't have another explanation for why this change here is required to find libwebpmux.lib .

bold84 committed:
updated version

dg0yt commented:
> What's even weirder is that even the zeroc-ice port (x64-windows-static) suddenly fails on your windows servers, although nothing was changed.

This is a baseline regression, also failing in the daily full builds: https://dev.azure.com/vcpkg/public/_build/results?buildId=70644&view=results

bold84 commented:
That you for letting me know. I figured that out by now 😊.
I was able to reproduce the issues and make everything build successfully, despite the regression.

I think that this was a good opportunity to fix a potential problem with the zeroc-ice port.

I will push the changes soon (still in bed).

dg0yt commented:
@bold84 There is already other work on zero-ice: #24169

bold84 committed:
Fix static crt issue

bold84 committed:
Fix Skia for arm64-windows

bold84 committed:
disable linux again

bold84 committed:
disable linux

bold84 committed:
update version

bold84 committed:
set v8 to fail on linux

bold84 committed:
skia doesn't support x86 builds with local MSVC installations

bold84 commented:
1. x86-windows has been disabled for skia. 
<img width="927" alt="Screen Shot 2022-04-17 at 17 18 09" src="https://user-images.githubusercontent.com/21118257/163710308-891f8593-5b26-4186-a3c6-70e1a7f912e9.png">

2. v8 has been disabled or expected to fail on linux
<img width="484" alt="Screen Shot 2022-04-17 at 17 19 16" src="https://user-images.githubusercontent.com/21118257/163710331-5b18ae62-7b86-4a62-ae14-30615ec955f7.png">

3. Skia fails to build on linux. It's looking for fontconfig/fontconfig.h which is present but isn't found. I will solve this problem later. 

4. zeroc-ice port has been changed so that the crt linkage and library linkage are now written directly to the project files before msbuild is called. @strega-nil please confirm that this solves the static crt issue you had.

dg0yt commented on scripts/ci.baseline.txt:
```diff
@@ -1406,6 +1402,11 @@ v8:x86-windows=fail
+v8:arm-uwp=fail
+v8:x64-osx=fail
+v8:x64-uwp=fail
+v8:x84-linux=fail
```
IIUC there is no PR CI left for v8.
And I guess from the headline that these four windows entries were intentionally listed separately.
Last not least this heavily overlaps with the `supports` field in v8 manifest.

bold84 replied:
I moved them together, so there aren't many different locations with information for the same port. 

I started updating v8, but soon got into difficulties. The build requires the breakpad python module. Removing that line caused a whole lot of other errors and since I originally just wanted to update gn and skia and because the current v8 version doesn't build with vs2022 anyways, I thought it might not hurt not caring too much about it.

How do you mean overlaps with the supports field?

bold84 replied:
vcpkg-tool-gn, alright... will add in a separate PR (or in this one?)

dg0yt replied:
> I moved them together, so there aren't many different locations with information for the same port.

AFAIU there is alphabetical group of generic issues, and a number of groups with specific, sometimes temporary, issues. So in general, "there aren't many different locations with information for the same port". 

> How do you mean overlaps with the supports field?

If a port (upstream) doesn't support a platform at all, it must be excluded via `supports` and doesn't need an entry in `ci.baseline.txt`.
Only if a platform is supported, but fails in CI, it should have a `fail` entry. Or a `skip` entry if  a port causes conflicts.
`fail` entries are still built in daily CI, to detect what unexpected pass.

bold84 replied:
> AFAIU there is alphabetical group of generic issues, and a number of groups with specific, sometimes temporary, issues. So in general, "there aren't many different locations with information for the same port".
Oh... I didn't know that. I will separate them again. 

> If a port (upstream) doesn't support a platform at all, it must be excluded via `supports` and doesn't need an entry in `ci.baseline.txt`. Only if a platform is supported, but fails in CI, it should have a `fail` entry. Or a `skip` entry if a port causes conflicts. `fail` entries are still built in daily CI, to detect unexpected pass.
Ahh... :-) Thank you for the explanation. I thought they have to be added to both locations (`supports` and `ci.baseline.txt`). I'll fix it. ☺️

dg0yt commented on scripts/ci.baseline.txt:
```diff
@@ -1186,7 +1186,7 @@ signalrclient:arm-uwp=fail
 skia:x64-linux=fail
 skia:x64-uwp=fail
-skia:x86-windows=fail
+skia:x86-windows=fail # skia doesn't support x86 builds with local MSVC installations
```
This is redundant, given the `supports` field in the manifest.
And I really don't understand what "local" is meant to indicate here.

bold84 replied:
It's a text I copied from their BUILD.gn file. 

See the screenshot:
https://github.com/microsoft/vcpkg/pull/24066#issuecomment-1100848079

JonLiu1993 replied:
if this port doesn't support x86-windows triplets, please add supports keyword in vcpkg.json instead of modify baseline.

dg0yt commented on ports/skia/portfile.cmake:
```diff
@@ -47,7 +61,7 @@ function(cmake_to_gn_list OUTPUT_ INPUT_)
     else()
         string(REPLACE ";" "\",\"" TEMP "${INPUT_}")
-        set(${OUTPUT_} "[\"${TEMP}\"]" PARENT_SCOPE)
+        set(${OUTPUT_} "[\"${TEMP}\"]" PARENT_SCOPE)    
```
Extra whitespace.

dg0yt commented on ports/skia/portfile.cmake:
```diff
@@ -94,25 +108,20 @@ set(_INCLUDE_DIR "${CURRENT_INSTALLED_DIR}/include")
-replace_skia_dep(libwebp "/include" "webp;webpdemux;webpdecoder;webpmux" "webp;webpdemux;webpdecoder;webpmux" "")
+replace_skia_dep(libwebp "/include" 
+    "webp,webpd;webpdemux,webpdemuxd;webpdecoder,webpdecoderd;libwebpmux,libwebpmuxd,webpmux,webpmuxd" 
+    "webp;webpdemux;webpdecoder;libwebpmux,webpmux" "")
```
Outdated. Update your webp installation.

dg0yt commented on ports/skia/portfile.cmake:
```diff
@@ -121,9 +130,10 @@ endif()
-    list(APPEND SKIA_PUBLIC_DEFINITIONS SK_METAL)
 endif()
 
+set(OPTIONS "${OPTIONS} target_cpu=\"${VCPKG_TARGET_ARCHITECTURE}\"")
```
Embedded quotes invite cross-platform issues.
Apart from that, `string(APPEND )` may make the appending a more explicit string operation.

bold84 replied:
I've tried several other ways... single quotes, no quotes... GN doesn't accept the architecture string without double quotes.

dg0yt commented on ports/skia/portfile.cmake:
```diff
@@ -174,45 +184,58 @@ vcpkg_install_gn(
 
 message(STATUS "Installing: ${CURRENT_PACKAGES_DIR}/include/${PORT}")
-file(COPY "${SOURCE_PATH}/include"
+file(COPY "${SOURCE_PATH}/include" 
```
Extra whitespace here and below.

dg0yt reviewed (commented):
Maybe it would be better to add a versioned `vcpkg-tool-gn` port instead of updating the `vcpkg_find_acquire_program`. Similar to `vcpkg-tool-meson` etc.

This would also avoid the world rebuilds which are triggered from touching that script.

bold84 committed:
review change requests

bold84 committed:
review changes

bold84 committed:
added vcpkg-tool-gn

bold84 committed:
Updated license fields and reverted changes to ci.baseline.txt

bold84 committed:
enable linux again :-)

bold84 committed:
Enabled additional backends

bold84 commented:
Because I assume that this PR triggers a "world build" with every commit, I will continue working on Skia here:

https://github.com/bold84/vcpkg/tree/update_skia_local

Once this PR is merged, I will open a new PR with the changes therein.

bold84 commented:
What distribution are these linux builds running on?

<img width="1028" alt="Screen Shot 2022-04-18 at 09 49 42" src="https://user-images.githubusercontent.com/21118257/163746617-d369cb26-ed71-4413-a4e4-e1524762fffb.png">

dg0yt commented:
> What distribution are these linux builds running on?

It shouldn't matter.
~~~
/usr/bin/ld: cannot find -lmcpp
collect2: error: ld returned 1 exit status
make: *** [Makefile:28: bin/slice2cpp] Error 1
~~~
Do you know that you can download the CI failure logs yourself, via the publised artifacts in Azure Pipelines?
And I wouldn't run builds as root user on Linux on my local system.

bold84 commented:
> > What distribution are these linux builds running on?
> 
> It shouldn't matter.
> 
> ```
> /usr/bin/ld: cannot find -lmcpp
> collect2: error: ld returned 1 exit status
> make: *** [Makefile:28: bin/slice2cpp] Error 1
> ```
> 
> Do you know that you can download the CI failure logs yourself, via the publised artifacts in Azure Pipelines? And I wouldn't run builds as root user on Linux on my local system.

Thank you. Yep, I know bout the failure logs. Unfortunately they are only available after all builds are done.

Thank you for the hint about running the build as root. It's a Docker container :-P

bold84 commented:
Well, the linux build was fine before. But something has changed and caused it to fail. There is another PR, that fixes the mcpp library name issue. I guess I have to wait until it's merged.

bold84 committed:
Fixes after merge conflicts

bold84 committed:
updated version

bold84 committed:
Increased port version for zeroc-ice and skia and added warning for building skia[dawn] feature on Linux.

bold84 committed:
don't install dawn when not built

bold84 commented:
> @bold84 , If this pr is ready for review please let me know, thanks!

It’s ready for review.

JonLiu1993 commented on ports/chromium-base/vcpkg.json:
```diff
@@ -1,8 +1,15 @@
   "name": "chromium-base",
-  "version-string": "86.0.4199.1",
+  "version": "86.0.4199.1",
   "port-version": 3,
```
```suggestion
  "port-version": 4,
```

JonLiu1993 commented on ports/crashpad/portfile.cmake:
```diff
@@ -3,9 +3,14 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 )
 
+vcpkg_find_acquire_program(PYTHON3)
+file(READ "${SOURCE_PATH}/.gn" GN_FILE_CONTENT)
+string(REPLACE "script_executable = \"python3\"" "script_executable = \"${PYTHON3}\"" GN_FILE_CONTENT ${GN_FILE_CONTENT})
+file(WRITE "${SOURCE_PATH}/.gn" ${GN_FILE_CONTENT})
```
```suggestion
vcpkg_replace_string("${SOURCE_PATH}/.gn" "script_executable = \"python3\"" "script_executable = \"${PYTHON3}\"")
```

bold84 replied:
Oh, I didn't know that function. Cool!

JonLiu1993 commented on ports/crashpad/vcpkg.json:
```diff
@@ -1,14 +1,19 @@
   "name": "crashpad",
-  "version-date": "2020-03-18",
+  "version-date": "2022-04-16",
   "port-version": 2,
```
```suggestion
```

JonLiu1993 commented on ports/skia/fontconfig.gn:
```diff
@@ -0,0 +1,10 @@
+   } else {
+     libs = @_LIBS_REL@
+   }
+ }
```
```suggestion
 }
 
```

JonLiu1993 commented on ports/skia/portfile.cmake:
```diff
@@ -1,10 +1,21 @@
 
+# Replace hardcoded python paths
+vcpkg_find_acquire_program(PYTHON3)
+file(READ "${SOURCE_PATH}/.gn" GN_FILE_CONTENT)
+string(REPLACE "script_executable = \"python3\"" "script_executable = \"${PYTHON3}\"" GN_FILE_CONTENT ${GN_FILE_CONTENT})
+file(WRITE "${SOURCE_PATH}/.gn" ${GN_FILE_CONTENT})
```
```suggestion
vcpkg_replace_string("${SOURCE_PATH}/.gn" "script_executable = \"python3\"" "script_executable = \"${PYTHON3}\"")
```

JonLiu1993 commented on ports/skia/portfile.cmake:
```diff
@@ -1,10 +1,21 @@
+string(REPLACE "script_executable = \"python3\"" "script_executable = \"${PYTHON3}\"" GN_FILE_CONTENT ${GN_FILE_CONTENT})
+file(WRITE "${SOURCE_PATH}/.gn" ${GN_FILE_CONTENT})
+
+file(READ "${SOURCE_PATH}/gn/toolchain/BUILD.gn" GN_FILE_CONTENT)
+string(REPLACE "command = \"$shell python" "command = \"$shell '${PYTHON3}'" GN_FILE_CONTENT ${GN_FILE_CONTENT})
+file(WRITE "${SOURCE_PATH}/gn/toolchain/BUILD.gn" ${GN_FILE_CONTENT})
```
```suggestion
vcpkg_replace_string("${SOURCE_PATH}/gn/toolchain/BUILD.gn" "command = \"$shell python" "command = \"$shell '${PYTHON3}'")
```

JonLiu1993 commented on ports/skia/vcpkg.json:
```diff
@@ -1,26 +1,23 @@
-  "version-date": "2020-05-18",
-  "port-version": 9,
+  "version-date": "2022-04-15",
+  "port-version": 10,
```
```suggestion
```

JonLiu1993 commented on ports/v8/vcpkg.json:
```diff
@@ -4,7 +4,8 @@
   "port-version": 3,
```
```suggestion
  "port-version": 4,
```

JonLiu1993 commented on versions/s-/skia.json:
```diff
@@ -1,8 +1,18 @@
 {
   "versions": [
     {
-      "git-tree": "cdf8cda1612a4770f3a01e35ae5da93654abc17d",
-      "version-date": "2020-05-18",
```
We should keep the history git-tree not change and add the new  "git-tree", please revert the history "git-tree" and only add a new one

JonLiu1993 commented on versions/s-/skia.json:
```diff
@@ -1,8 +1,18 @@
+      "version-date": "2022-04-15",
+      "port-version": 10
+    },
+    {
+      "git-tree": "0aaf1c1264354e97be0edfe5b017eddfcf6a66bc",
+      "version-date": "2022-04-15",
+      "port-version": 9
+    },
+    {
+      "git-tree": "68e77104f54fa164b2890ae0eb860315b8cf318c",
+      "version-date": "2022-04-10",
```
```suggestion

```

JonLiu1993 commented on versions/s-/skia.json:
```diff
@@ -1,8 +1,18 @@
     {
-      "git-tree": "cdf8cda1612a4770f3a01e35ae5da93654abc17d",
-      "version-date": "2020-05-18",
+      "git-tree": "6d3891b5706edb3b2d3496fa8c0fd32978a8cd78",
+      "version-date": "2022-04-15",
+      "port-version": 10
```
please revert the history "git-tree" and only add this

JonLiu1993 commented on versions/z-/zeroc-ice.json:
```diff
@@ -1,7 +1,12 @@
 {
   "versions": [
     {
-      "git-tree": "f32e4c338e25d36018095348e632c794b13fc513",
```
same as above , We should keep the history git-tree not change and add the new "git-tree", please revert the history "git-tree" and only add a new one

bold84 replied:
I don't fully understand what exactly should be done and why. Please elaborate, so I can avoid this problem in the future.

bold84 replied:
Nevermind... I think I got it... port-version should be 9 though, if I understand correctly.

bold84 committed:
Update ports/chromium-base/vcpkg.json

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

bold84 committed:
Update ports/crashpad/portfile.cmake

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

bold84 committed:
Update ports/crashpad/vcpkg.json

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

bold84 committed:
Update ports/skia/portfile.cmake

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

bold84 committed:
fixed git-tree issue

bold84 committed:
Fixed versioning / git-tree

bold84 commented:
@JonLiu1993 Okay... I see now how much I f...ed up the versioning / git-tree. 
I think I have corrected everything to the values they should have. Please have a look.

JonLiu1993 commented on ports/crashpad/vcpkg.json:
```diff
@@ -1,14 +1,19 @@
-  "version-date": "2020-03-18",
-  "port-version": 2,
+  "version-date": "2022-04-16",
+  "port-version": 3,
```
```suggestion
```

bold84 replied:
Why is port-version removed?

JonLiu1993 replied:
Because when we update the port , we should remove the port-version, you can refer to [this ](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-port-version-field-in-the-manifest-file-of-any-modified-ports)

bold84 replied:
Thank you!! Now, understand.

JonLiu1993 commented on ports/skia/vcpkg.json:
```diff
@@ -1,26 +1,23 @@
   "name": "skia",
-  "version-date": "2020-05-18",
+  "version-date": "2022-04-15",
   "port-version": 9,
```
```suggestion
```

JonLiu1993 commented on versions/s-/skia.json:
```diff
@@ -1,8 +1,8 @@
     {
-      "git-tree": "cdf8cda1612a4770f3a01e35ae5da93654abc17d",
-      "version-date": "2020-05-18",
+      "git-tree": "3766e50c9d64fceedf7f7eadb2b8382907eed985",
+      "version-date": "2022-04-15",
       "port-version": 9
     },
```
```suggestion
    {
      "git-tree": "cdf8cda1612a4770f3a01e35ae5da93654abc17d",
      "version-date": "2020-05-18",
      "port-version": 9
    },
```

bold84 replied:
What am I understanding wrong?
The last change to Skia that was merged was two months ago; port-version: 8

From what I can understand, the next version should represent the changes of this PR and therefore be port-version: 9

What am I getting wrong?

JonLiu1993 replied:
Currently, the skia in vcpkg port-version is port-version: 9, [this ](https://github.com/microsoft/vcpkg/commit/99346bb6926e85d93e4aad330bf28cce4a18051b)is the last commit

bold84 replied:
Ops… I see 👍

bold84 committed:
Update ports/skia/portfile.cmake

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

bold84 committed:
Update ports/skia/vcpkg.json

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

bold84 committed:
Update ports/crashpad/vcpkg.json

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

bold84 committed:
updated ci.ci.baseline.txt

bold84 committed:
fixed versions

bold84 commented:
<img width="607" alt="Screen Shot 2022-04-20 at 18 15 17" src="https://user-images.githubusercontent.com/21118257/164218924-76a9d099-4096-4e73-93f7-4ffea97fbd13.png">

https://github.com/microsoft/vcpkg/actions/runs/2195232664

What's wrong?

JackBoosY commented:
```
Error: Unhandled error: HttpError: Unknown error: {"count":11,"value":"Uri expired"}
```
@BillyONeal  Do you know why?

bold84 commented:
/azp run

JackBoosY commented:
Will rerun the test after other tests finished.

JackBoosY commented:
Depends on https://github.com/microsoft/vcpkg/pull/24267.

bold84 commented:
I have learned a lot from this PR 🤓. Thank you for your review and recommendations.

Besides other things, next time I'll plan changes to core files a bit better.

JackBoosY commented:
/azp run

bold84 commented:
Hmm… There are several things I could improve. But I’ll create new PRs for them.

JackBoosY commented on ports/skia/portfile.cmake:
```diff
@@ -19,10 +26,27 @@ function(checkout_in_path PATH URL REF)
     file(REMOVE_RECURSE "${DEP_SOURCE_PATH}")
 endfunction()
 
+function(checkout_in_path_with_patch PATH URL REF PATCH)
```
Defined this function but not use it?

bold84 replied:
It was part of an unfinished PR on here, in which one of the dependencies required a patch. With the newer ref I used, that patch was not required anymore.

I’m aware this function is currently not used, but it might be used in the future. So, I felt it doesn’t hurt to leave it there.

JackBoosY commented on ports/skia/skiaConfig.cmake.in:
```diff
@@ -1,12 +1,9 @@
 add_library(skia INTERFACE)
```
Did you test the usage?

bold84 replied:
No, I just took the changes from the previously mentioned PR. 
I know that this file of the previous port version did not work (tested), so it couldn’t get worse than that. I will improve this file, but I prefer this to be in a separate PR, as one round trip (build) of this PR takes several hours due to the involvement of core files.

strega-nil committed:
versions + space changes

strega-nil-ms commented:
rebased and fixed spaces; gonna wait for green one last time :)

Thomas1664 commented on ports/crashpad/vcpkg.json:
```diff
@@ -1,14 +1,18 @@
   "name": "crashpad",
-  "version-date": "2020-03-18",
-  "port-version": 2,
+  "version-date": "2022-04-16",
```
```suggestion
  "version-date": "2022-04-12",
```
Please use the actual commit date

dg0yt replied:
> Please use the actual commit date

Wait. I'm not convinced this is the de-facto policy.
(I don't see it written, except for [old hints about `version-string`](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/manifest-files.md#version-string):
`we use the date that the commit was accessed by you`)

Thomas1664 replied:
I saw several reviews where it was asked to use commit date as version-date. However, after some research I saw both policies.

BillyONeal commented:
> rebased and fixed spaces; gonna wait for green one last time :)

I didn't review; I just merged green build based on this :)

njakob commented on ports/crashpad/portfile.cmake:
```diff
@@ -119,6 +122,9 @@ file(REMOVE_RECURSE
 configure_file("${CMAKE_CURRENT_LIST_DIR}/crashpadConfig.cmake.in"
         "${CURRENT_PACKAGES_DIR}/share/${PORT}/crashpadConfig.cmake" @ONLY)
 
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/include/${PORT}/build")
```
I'm currently running into the following error while using the latest version of crashpad (`2022-04-16`) on Windows. Can this line be responsible for the missing files and is this an intended behavior?

> Error: %REDACTED%\build\windows-release\vcpkg_installed\x64-windows\include\crashpad\base/files/file_path.h(110): fatal error C1083: Cannot open include file: 'build/build_config.h': No such file or directory

I'm not sure if it's related to https://github.com/microsoft/vcpkg/issues/24510 or if it deserves its own issue?

CC @bold84

njakob replied:
I created a new issue regarding this problem https://github.com/microsoft/vcpkg/issues/25066
